src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -428,6 +428,17 @@ public void Initialize(ITelemetry telemetry)
                 var version = LazyInitializer.EnsureInitialized(ref this.sdkVersion, this.GetSdkVersion);
                 telemetry.Context.Internal.SdkVersion = version;
             }
+
+            // set RoleInstance and NodeName to the machine name if they're not initialized yet
+            if (string.IsNullOrEmpty(telemetry.Context.Internal.NodeName))
+            {
+                telemetry.Context.Internal.NodeName = PlatformSingleton.Current.GetMachineName();
```

Comment:
Also should we try to use roleINstance first and then fall back to the computer name? Or we do not want to trust the customer's choice or roleInstance?